src/components/core/geoChart/GeoChartOptionsWrapper.tsx

```diff
@@ -13,37 +13,50 @@ import { getColorStrategy } from "../../../helpers/geoChart/colorStrategy";
 import { isMappingHeaderAttributeItem } from "../../../interfaces/MappingHeader";
 import { BaseVisualization } from "../base/BaseVisualization";
 import { GeoChartInner, IGeoChartInnerOptions, IGeoChartInnerProps } from "./GeoChartInner";
+import { fixEmptyHeaderItems } from "../base/utils/fixEmptyHeaderItems";
 
 export class GeoChartOptionsWrapper extends BaseVisualization<IGeoChartInnerProps, null> {
     public static defaultProps: Partial<IGeoChartInnerProps> = GeoChartInner.defaultProps;
-
     public renderVisualization(): JSX.Element {
+        const {
+            execution: { executionResult },
+        } = this.props;
+        const fixedExecutionResult = fixEmptyHeaderItems(executionResult, this.getEmptyHeaderString());
+
+        const newProps = {
+            ...this.props,
+            execution: { ...this.props.execution, executionResult: fixedExecutionResult },
+        };
         const {
             config: { mdObject },
             execution,
             onDataTooLarge,
-        } = this.props;
+        } = newProps;
+
         const buckets = getGeoBucketsFromMdObject(mdObject);
         const geoData: IGeoData = getGeoData(buckets, execution);
```

Comment:
should it use the new `execution`?
`const geoData: IGeoData = getGeoData(buckets, newProps.execution);`

Reply:
execution was got from newProps.